src/lib/Microsoft.Health.Events/EventHubProcessor/EventProcessorClientFactory.cs

```diff
@@ -23,27 +24,24 @@ public EventProcessorClient CreateProcessorClient(BlobContainerClient blobContai
             if (options.ServiceManagedIdentityAuth)
```

Comment:
Nit, would that make sense to add an enum property in the EventProcessorClientFactoryOptions to explicitly say the type of connection/authentication? The if-else seems able to handle overriding logic flow in the code itself but I'm not sure whether that's a valid use case.  Other than that, I think a single property to specify which type of authentication would make code cleaner.

Reply:
If we have a known set an enum seems warrented.

Reply:
Is the idea to have a "authentication type" property in EventProcessorClientFactoryOptions that is set to the type of connection/authentication and is calculated based on what options (connection string, event hub mi settings, other) are provided? I agree this could make the code cleaner, but just making sure I understand the comment.

Or do we want to user to explicitly set this new "authentication type" property, in addition to supplying the required connection options properties.

Reply:
I was thinking the second one in here, as user, I will have to tell this factory explicitly which auth option I want to use by setting the enum field.

Reply:
Thanks for making the changes! They look good to me.